File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/cli/FetchOrientation.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.spark.sql.hive.thriftserver.cli

import org.apache.spark.service.cli.thrift.TFetchOrientation

trait FetchOrientation {
  def toTFetchOrientation: TFetchOrientation
}

object FetchOrientation {

  case object FETCH_NEXT extends FetchOrientation {
    override val toTFetchOrientation: TFetchOrientation = TFetchOrientation.FETCH_NEXT
  }

  case object FETCH_PRIOR extends FetchOrientation {
    override val toTFetchOrientation: TFetchOrientation = TFetchOrientation.FETCH_PRIOR
  }

  case object FETCH_RELATIVE extends FetchOrientation {
    override val toTFetchOrientation: TFetchOrientation = TFetchOrientation.FETCH_RELATIVE
  }

  case object FETCH_ABSOLUTE extends FetchOrientation {
    override val toTFetchOrientation: TFetchOrientation = TFetchOrientation.FETCH_ABSOLUTE
  }

  case object FETCH_FIRST extends FetchOrientation {
    override val toTFetchOrientation: TFetchOrientation = TFetchOrientation.FETCH_FIRST
  }

  case object FETCH_LAST extends FetchOrientation {
    override val toTFetchOrientation: TFetchOrientation = TFetchOrientation.FETCH_LAST
  }

  def getFetchOrientation(tFetchOrientation: TFetchOrientation): FetchOrientation =
    tFetchOrientation match {
      case FETCH_FIRST.toTFetchOrientation => FETCH_FIRST
      case _ => FETCH_NEXT
    }
}

File: sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/FetchOrientation.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.hive.service.cli;

import org.apache.hive.service.rpc.thrift.TFetchOrientation;

/**
 * FetchOrientation.
 *
 */
public enum FetchOrientation {
  FETCH_NEXT(TFetchOrientation.FETCH_NEXT),
  FETCH_PRIOR(TFetchOrientation.FETCH_PRIOR),
  FETCH_RELATIVE(TFetchOrientation.FETCH_RELATIVE),
  FETCH_ABSOLUTE(TFetchOrientation.FETCH_ABSOLUTE),
  FETCH_FIRST(TFetchOrientation.FETCH_FIRST),
  FETCH_LAST(TFetchOrientation.FETCH_LAST);

  private TFetchOrientation tFetchOrientation;

  FetchOrientation(TFetchOrientation tFetchOrientation) {
    this.tFetchOrientation = tFetchOrientation;
  }

  public static FetchOrientation getFetchOrientation(TFetchOrientation tFetchOrientation) {
    for (FetchOrientation fetchOrientation : values()) {
      if (tFetchOrientation.equals(fetchOrientation.toTFetchOrientation())) {
        return fetchOrientation;
      }
    }
    // TODO: Should this really default to FETCH_NEXT?
    return FETCH_NEXT;
  }

  public TFetchOrientation toTFetchOrientation() {
    return tFetchOrientation;
  }
}
